### PR TITLE
Feature/movie tile responsive patch2

### PR DESCRIPTION
**Pull request by damianmichalowski**

### CODE REVIEW

**Comment by JamieShifter** [comment]
Houston, we have a problem at iPhone5:
image

I'd suggest figuring out how to dot out the longer titles

File: src/features/MovieList/styled.js [hunk 1]
```diff
@@ -1,25 +1,29 @@
...
+    margin: 0 auto;
     display: flex;
     flex-direction: column;
 `;
 
-export const Title = styled.h1`
+export const Title = styled.h2`
     font-family: 'Poppins';
     font-style: normal;
     font-weight: 600;
```
Comment (JamieShifter) [inline src/features/MovieList/styled.js]: I think we're getting most of it by default already, maybe it's not needed? (Unless we want to be explicit)

**Review by JamieShifter — APPROVED** [review]
I think you did a great job, it looks very similar to figma. Only problem I see for now is the longer titles, especially on mobile devices, might be worth looking into that, today we'll decide if we should do that now or later;)
🥇

File: src/features/MovieList/styled.js (hunk 1 above)
Comment (damianmichalowski) [reply src/features/MovieList/styled.js]: you right, thanks, i fix it

**Comment by damianmichalowski** [comment]
[Movie-list fixes](https://trello.com/c/DYdsba8U/25-movie-list-fixes)

**Comment by damianmichalowski** [comment]
i added responsive title "Popular Movies"  check it out

File: src/features/MovieList/styled.js [hunk 2]
```diff
@@ -1,25 +1,33 @@
...
     flex-direction: column;
 `;
 
-export const Title = styled.h1`
-    font-family: 'Poppins';
-    font-style: normal;
-    font-weight: 600;
+export const Title = styled.h2`
     font-size: 36px;
-    line-height: 120%;
```
Comment (Marcin-Malek) [inline src/features/MovieList/styled.js:16]: Why did you remove line-height? I know the UI design in Figma is not thorough, but this was not redundant or interruptive.

File: src/features/MovieList/styled.js [hunk 3]
```diff
@@ -1,25 +1,33 @@
 import styled from "styled-components";
 
 export const Content = styled.section`
-    max-width: 1368px;
-    margin: auto;
+    max-width: 1400px;
+    padding:0 16px;
+    margin: 0 auto;
     display: flex;
     flex-direction: column;
```
Comment (Marcin-Malek) [inline src/features/MovieList/styled.js]: You don't need `display: flex;` here now.

File: src/features/MovieList/styled.js [hunk 4]
```diff
@@ -1,25 +1,33 @@
...
-    padding: 56px 16px 24px 16px;
+    padding: 56px 0 24px;
+
+    @media (max-width: 1023px ) {
+        font-size:3.5vw;
+    }
+
+    @media (max-width: 500px) {
+        font-size:17.3px;
+    }
```
Comment (Marcin-Malek) [inline src/features/MovieList/styled.js]: It looks cool, although we need to discuss breakpoints. I think it is better in terms of clarity if we specified few of them in theme.js instead of creating new ones with different values in every file.

File: src/features/MovieList/styled.js [hunk 5]
```diff
@@ -1,25 +1,33 @@
...
 `;
 
 export const Wrapper = styled.div`
     display: grid;
```
Comment (Marcin-Malek) [inline src/features/MovieList/styled.js]: Consider changing this div to `<ul>` and MovieTile component to `<li>`. What we are presenting in a MovieList component is in fact a list of movies, an unordered one in this case. 

[Some Informations about HTML lists](https://www.w3.org/TR/html401/struct/lists.html#h-10.1)

File: src/features/MovieList/MovieTile/styled.js [hunk 6]
```diff
@@ -66,27 +68,31 @@ export const Year = styled.p`
...
     flex-wrap: wrap;
     gap: 8px;
+    padding: 0;
+    list-style: none;
+    margin-bottom: 0;
 `;
 
-export const Tag = styled.p`
+export const Tag = styled.li`
     margin: 0px;
```
Comment (Marcin-Malek) [inline src/features/MovieList/MovieTile/styled.js]: You dont need units when setting margin to 0

File: src/features/MovieList/MovieTile/styled.js [hunk 7]
```diff
@@ -66,27 +68,31 @@ export const Year = styled.p`
...
 
-export const TagsContainer = styled.div`
+export const TagsContainer = styled.ul`
     margin-top: 8px;
     display: flex;
     flex-wrap: wrap;
     gap: 8px;
+    padding: 0;
+    list-style: none;
+    margin-bottom: 0;
```
Comment (Marcin-Malek) [inline src/features/MovieList/MovieTile/styled.js]: you could use `margin: 8px 0 0;`

File: src/features/MovieList/MovieTile/styled.js [hunk 8]
```diff
@@ -66,27 +68,31 @@ export const Year = styled.p`
...
         font-size: 10px;
     }
 `;
 
 export const RatingContainer = styled.div`
-    margin-top: 39px;
+    margin-top: 14px;
     display: flex;
     align-items: center;
     align-content: flex-end;
```
Comment (Marcin-Malek) [inline src/features/MovieList/MovieTile/styled.js:103]: align-content does not do anything here.

File: src/features/MovieList/MovieTile/styled.js [hunk 9]
```diff
@@ -104,7 +110,7 @@ export const StyledStar = styled(Star)`
     }
 `;
 
-export const Rate = styled.p`
+export const Rate = styled.span`
     margin: 0px;
```
Comment (Marcin-Malek) [inline src/features/MovieList/MovieTile/styled.js]: Same here, units not needed, look through the file and check if it appears somewhere else.

File: src/features/MovieList/MovieTile/index.js [hunk 10]
```diff
@@ -14,12 +14,16 @@ import {
...
 export const MovieTile = ({ title, year, tags, rate, votes }) => {
     return (
         <Content>
-            <Img src={Poster} />
+            <ImgWrapper>
+                <Img src={Poster} />
+            </ImgWrapper>
+
             <Wrapper>
                 <DescriptionContainer>
```
Comment (Marcin-Malek) [inline src/features/MovieList/MovieTile/index.js]: DescriptionContainer component is not needed. You could remove it from styled.js and imports and replace it with `<div>` on this line

File: src/features/MovieList/MovieTile/styled.js [hunk 11]
```diff
@@ -66,27 +68,31 @@ export const Year = styled.p`
...
+    
     @media (max-width: ${({ theme }) => theme.breakpoint.mobile}px) {
         padding: 4px 8px;
         font-size: 10px;
     }
 `;
 
 export const RatingContainer = styled.div`
-    margin-top: 39px;
+    margin-top: 14px;
```
Comment (Marcin-Malek) [inline src/features/MovieList/MovieTile/styled.js]: This value is different in Figma, is this bigger on purpose?

**Review by Marcin-Malek — COMMENTED** [review]
Here are some suggestions and questions from me.

File: src/features/MovieList/MovieTile/styled.js [hunk 12]
```diff
@@ -66,27 +68,31 @@ export const Year = styled.p`
     }
 `;
 
-export const TagsContainer = styled.div`
+export const TagsContainer = styled.ul`
     margin-top: 8px;
     display: flex;
     flex-wrap: wrap;
     gap: 8px;
```
Comment (damianmichalowski) [inline src/features/MovieList/MovieTile/styled.js]: If someone want to use our browser on 176px this solve problem going over the edge of container.
i know There are no such screens 176px xd, but if there is a 100% responsive option, why not

File: src/features/MovieList/MovieTile/styled.js [hunk 13]
```diff
@@ -66,27 +68,31 @@ export const Year = styled.p`
...
+    padding: 0;
```
Comment (damianmichalowski) [inline src/features/MovieList/MovieTile/styled.js]: this mainly makes the rating stick to the bottom

File: src/features/MovieList/styled.js (hunk 2 above)
Comment (damianmichalowski) [reply src/features/MovieList/styled.js:16]: nothing change

File: src/features/MovieList/styled.js (hunk 3 above)
Comment (damianmichalowski) [reply src/features/MovieList/styled.js]: fixed

File: src/features/MovieList/styled.js (hunk 4 above)
Comment (damianmichalowski) [reply src/features/MovieList/styled.js]: i can add breakpoint into theme, finally the size is 536px, so it can be named "specificDevice" or sth like this

File: src/features/MovieList/styled.js (hunk 5 above)
Comment (damianmichalowski) [reply src/features/MovieList/styled.js]: okey it makes sense i will change this

File: src/features/MovieList/MovieTile/styled.js (hunk 6 above)
Comment (damianmichalowski) [reply src/features/MovieList/MovieTile/styled.js]: fixed

File: src/features/MovieList/MovieTile/styled.js (hunk 7 above)
Comment (damianmichalowski) [reply src/features/MovieList/MovieTile/styled.js]: margin: 8px 0 0;         is more optimalized than  margin-top ? 
i can changed this but what purpose ?

File: src/features/MovieList/MovieTile/styled.js (hunk 8 above)
Comment (damianmichalowski) [reply src/features/MovieList/MovieTile/styled.js:103]: align-content change everythink look again delete this and paste its very important line of code, 
if it wasn't there the rating will go up, as we know we want rating stick to bottom

File: src/features/MovieList/MovieTile/index.js (hunk 10 above)
Comment (damianmichalowski) [reply src/features/MovieList/MovieTile/index.js]: problem risolve, the descriptionContainer is deleted now

File: src/features/MovieList/MovieTile/styled.js (hunk 11 above)
Comment (damianmichalowski) [reply src/features/MovieList/MovieTile/styled.js]: i think 12 px like on figma is to close , but i can change to 12px maybe later we will changed it

File: src/features/MovieList/MovieTile/styled.js (hunk 7 above)
Comment (Marcin-Malek) [reply src/features/MovieList/MovieTile/styled.js]: You replaced only margin-top with `margin: 8px 0 0;` what i meant you should do is replace margin-top and margin-bottom with one margin shorthand. Using shorthands is very convenient and in my opinion specific attributes like margin-top should be used only if we want to overwrite just one property. This way reading code is easier and more elegant. It is purely esthetics and time-saving matter just as not specifying units after setting property as 0. Importance of this is arguable since it provides next to none optimalization improvement, however if you ask me it is not only about if the solution is working but also if it is intuitive and coherent.

File: src/features/MovieList/styled.js (hunk 5 above)
Comment (Marcin-Malek) [reply src/features/MovieList/styled.js]: Also this link has some information about definition lists you asked me on sunday daily. If you are curious about detailed information you might find it in this resource.

File: src/features/MovieList/MovieTile/styled.js (hunk 8 above)
Comment (Marcin-Malek) [reply src/features/MovieList/MovieTile/styled.js:103]: Check it once again, in my opinion it does not do anything. You probably thought of `justify-content: space-between;` property in Wrapper component, this is RatingContainer.

no-align-content
align-content

Also align-content has no effect on single line flex containers. In the time i made this comment this had default flex-wrap which is no-wrap - that meant it had to be single line. Here is an [align-content documentation](https://developer.mozilla.org/en-US/docs/Web/CSS/align-content). Besides do you think `flex-wrap: wrap` you added is needed here?

File: src/features/MovieList/MovieTile/styled.js (hunk 12 above)
Comment (Marcin-Malek) [reply src/features/MovieList/MovieTile/styled.js]: You are completely right, if we can make it better and working on more screens, possibly also not existing screens we should pursue it. Some people divide their screens into more applications so it is theoretically possible that someone would want to browse our page in browser that is shrinked so that the viewport is 176px wide. However im not sure what are you referencing to. Do you mean `flex-wrap: wrap;` is responsible for this responsiveness?

File: src/features/MovieList/styled.js (hunk 2 above)
Comment (Marcin-Malek) [reply src/features/MovieList/styled.js:16]: It surely does increase line-height a bit. Have you checked it?

File: src/features/MovieList/MovieTile/styled.js (hunk 8 above)
Comment (Marcin-Malek) [reply src/features/MovieList/MovieTile/styled.js:103]: Okay, i pulled your latest changes, I guess we need it now.

File: src/features/MovieList/styled.js (hunk 4 above)
Comment (Marcin-Malek) [reply src/features/MovieList/styled.js]: I might be wrong, but i think we should stick to real devices instead of our layout when we are specifying breakpoints in theme.js. The idea of adding breakpoints in theme.js is to use the same breakpoints in every component, not adding breakpoints we find ok for our layout to theme.js. Take a look at navigation bar, it has a breakpoint at 860pxs. Wouldn't it be better if both navigation and tiles changed at the same breakpoint?

File: src/features/MovieList/MovieTile/styled.js [hunk 14]
```diff
@@ -1,125 +1,158 @@
...
     line-height: 130%;
 
-    @media (max-width: ${({ theme }) => theme.breakpoint.mobile}px) {
+    @media (max-width: ${({ theme }) => theme.breakpoint.tablet}px) {
         font-size: 16px;
+        margin: 0 0 4px;
+    }
+
+    @media (max-width: ${({ theme }) => theme.breakpoint.mobile}px){
+        font-size: 4vw;
```
Comment (Marcin-Malek) [inline src/features/MovieList/MovieTile/styled.js:53]: If you are specifying font-size in vw units you should also specify margins and paddings in vw units. It does not look good when font is getting smaller, while the spacings remain the same. In order to resolve this issue you could either get rid of those vw units overall or add all of the margins and paddings in vw units also. I know i have done this with all the margins in my latest branch for backdrop description, but setting up all of the components like that is a huge overkill and definitely too time consuming. Also we should carefully use vw units, and by carefully i mean when it really is a good idea.

File: src/features/MovieList/MovieTile/styled.js [hunk 15]
```diff
@@ -1,125 +1,158 @@
...
-
-    @media (max-width: ${({ theme }) => theme.breakpoint.mobile}px) {
+    
+    @media (max-width: ${({ theme }) => theme.breakpoint.tablet}px){
         gap: 8px;
-        margin-top: 12px;
+    }
+
+    @media (min-width: ${({ theme }) => theme.breakpoint.tablet}px){
+        flex: 1;
```
Comment (Marcin-Malek) [inline src/features/MovieList/MovieTile/styled.js]: You dont need two media queries here. Just set `flex: 1;` for every media and then `flex: 0;` in `@media (max-width: ${({ theme }) => theme.breakpoint.tablet}px)`

File: src/features/MovieList/MovieTile/styled.js [hunk 16]
```diff
@@ -1,125 +1,158 @@
...
+    @media (max-width: ${({ theme }) => theme.breakpoint.mobile}px){
+        font-size: 3vw;
+    }
 `;
 
-export const Votes = styled.p`
-    margin: 0px;
+export const Votes = styled.span`
     font-weight: 400;
+    font-size: 16px;
```
Comment (Marcin-Malek) [inline src/features/MovieList/MovieTile/styled.js]: `font-weight: 400;` and `font-size: 16px;` are default `<span>` values.

File: src/features/MovieList/MovieTile/styled.js [hunk 17]
```diff
@@ -1,125 +1,158 @@
...
+        width: 3vw;
+        height: 3vw;
     }
 `;
 
-export const Rate = styled.p`
-    margin: 0px;
+export const Rate = styled.span`
     font-weight: 600;
     font-size: 16px;
```
Comment (Marcin-Malek) [inline src/features/MovieList/MovieTile/styled.js]: this font-size is a default value

File: src/features/MovieList/MovieTile/styled.js [hunk 18]
```diff
@@ -1,125 +1,158 @@
...
+    @media (min-width: ${({ theme }) => theme.breakpoint.tablet}px){
+        flex: 1;
     }
 `;
 
 export const StyledStar = styled(Star)`
-    @media (max-width: ${({ theme }) => theme.breakpoint.mobile}px) {
+    width: 24px;
+    height: 24px;
+    fill: none;
```
Comment (Marcin-Malek) [inline src/features/MovieList/MovieTile/styled.js]: Do we need `fill: none;` here

File: src/features/MovieList/MovieTile/styled.js [hunk 19]
```diff
@@ -1,125 +1,158 @@
...
+
+    @media (max-width: ${({ theme }) => theme.breakpoint.mobile}px){
+        font-size: 4vw;
     }
 `;
 
-export const Year = styled.p`
+export const Year = styled.span`
     margin: 0;
     font-size: 16px;
```
Comment (Marcin-Malek) [inline src/features/MovieList/MovieTile/styled.js]: This is default font-size

File: src/features/MovieList/MovieTile/styled.js (hunk 7 above)
Comment (Marcin-Malek) [reply src/features/MovieList/MovieTile/styled.js]: If you want to check how margin shorthands work here are a [few examples](https://developer.mozilla.org/en-US/docs/Web/CSS/Margin#more_examples).

File: src/features/MovieList/styled.js (hunk 4 above)
Comment (damianmichalowski) [reply src/features/MovieList/styled.js]: I can't change it, if we want this cool smooth responsive beautiful effect

File: src/features/MovieList/styled.js (hunk 2 above)
Comment (damianmichalowski) [reply src/features/MovieList/styled.js:16]: it is so much little different, so I decide to remove this line high and have less code, I can add this if you want to stack 1px to 1px to  Figma

File: src/features/MovieList/styled.js (hunk 5 above)
Comment (damianmichalowski) [reply src/features/MovieList/styled.js]: okay I understood now, thanks

File: src/features/MovieList/MovieTile/styled.js (hunk 7 above)
Comment (damianmichalowski) [reply src/features/MovieList/MovieTile/styled.js]: I didn't noticed I also have a margin-bottom there, sorry, fixed.

File: src/features/MovieList/MovieTile/styled.js (hunk 8 above)
Comment (damianmichalowski) [reply src/features/MovieList/MovieTile/styled.js:103]: if you want space between we need an extra container, better is align content I show you on daily, maybe our browsers work differently because on mine it looks different

File: src/features/MovieList/MovieTile/styled.js (hunk 9 above)
Comment (damianmichalowski) [reply src/features/MovieList/MovieTile/styled.js]: fixed

File: src/features/MovieList/MovieTile/styled.js (hunk 12 above)
Comment (damianmichalowski) [reply src/features/MovieList/MovieTile/styled.js]: yes flex-wrap making this super responsive

File: src/features/MovieList/MovieTile/styled.js (hunk 15 above)
Comment (damianmichalowski) [reply src/features/MovieList/MovieTile/styled.js]: ok thanks, fixed

File: src/features/MovieList/MovieTile/styled.js (hunk 16 above)
Comment (damianmichalowski) [reply src/features/MovieList/MovieTile/styled.js]: fixed

File: src/features/MovieList/MovieTile/styled.js (hunk 17 above)
Comment (damianmichalowski) [reply src/features/MovieList/MovieTile/styled.js]: fixed

File: src/features/MovieList/MovieTile/styled.js (hunk 18 above)
Comment (damianmichalowski) [reply src/features/MovieList/MovieTile/styled.js]: you right

File: src/features/MovieList/MovieTile/styled.js (hunk 19 above)
Comment (damianmichalowski) [reply src/features/MovieList/MovieTile/styled.js]: fixed

File: src/features/MovieList/MovieTile/styled.js (hunk 14 above)
Comment (damianmichalowski) [reply src/features/MovieList/MovieTile/styled.js:53]: I don't  see a problem here, I check this scalin and it looks great, you can show on daily what you mean

File: src/features/MovieList/styled.js (hunk 4 above)
Comment (Marcin-Malek) [reply src/features/MovieList/styled.js]: I can't agree with you. My point is, we can have smooth responsive effect with different breakpoints.

File: src/features/MovieList/MovieTile/styled.js (hunk 8 above)
Comment (Marcin-Malek) [reply src/features/MovieList/MovieTile/styled.js:103]: > if you want space between we need an extra container, better is align content I show you on daily, maybe our browsers work differently because on mine it looks different

No, its just at the time i made the comment i haven't got your latest changes. align-content is ok with your current solution.
